src/argprocessing.cpp

```diff
@@ -1310,7 +1310,8 @@ process_args(Context& ctx)
   if (config.run_second_cpp()) {
     extra_args_to_hash.push_back(state.dep_args);
   }
-  if (state.hash_full_command_line) {
+  if (state.hash_full_command_line
+      && !ctx.config.sloppiness().is_enabled(core::Sloppy::full_command_line)) {
```

Comment:
I would prefer it if you check the sloppiness option when setting the flag instead:

https://github.com/ccache/ccache/blob/6f9c35810004832751465fd8f4d88a4090c02e32/src/argprocessing.cpp#L805-L807